src/main/java/com/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.UserDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
```

Comment:
Let's put it inside try block

Reply:
what do you mean? we've agreed on not putting it inside try-with-resources block, and leave it out here because we need to have the ability to rollback the transaction